File: frontend/lib/types/notification.ts
export interface Notification {
  id: string;
  user_id: string;
  type: "new_chapter" | "reply_to_comment" | "comment_liked" | "gift_received";
  payload: Record<string, unknown>;
  read_at: string | null;
  created_at: string;
}

export interface UnreadCount {
  count: number;
}

File: backend/app/models/notification.py
from datetime import datetime
from typing import Any, Optional

from pydantic import BaseModel


class NotificationPublic(BaseModel):
    id: str
    user_id: str
    type: str
    payload: dict[str, Any]
    read_at: Optional[datetime] = None
    created_at: datetime


class UnreadCount(BaseModel):
    count: int
